### PR TITLE
Assignment 3

### PR DESCRIPTION
**Pull request by AvaEN**

### CODE REVIEW

File: assignment3.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Ava N.
+June 30, 2016
+
+Assignment 3
+
+1. The code written was invalid because the printf statement should look like this: printf("%d\n",i);
```
Comment (oldclesleycode) [inline assignment3.txt:6]: Correct! You'd also need to initialize i to a number greater than 0 beforehand for the program to properly run.

File: assignment3.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+2. i=(int) should be before the for loop. Also, the statement should look like this: int i;
```
Comment (oldclesleycode) [inline assignment3.txt:7]: Not quite! Check again and see if you can find the two things wrong with this code snippet! If you still can't find them, let me know!

File: assignment3.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+3. i.2^n
```
Comment (oldclesleycode) [inline assignment3.txt:8]: O(n^2) actually! The for loops are nested so you'd multiply their time complexities, so n*n = n^2

File: assignment3.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+   ii. 0(n^4)
```
Comment (oldclesleycode) [inline assignment3.txt:9]: Correct! Give an explanation though

File: assignment3.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+   iii.0(n)
```
Comment (oldclesleycode) [inline assignment3.txt:10]: Correct! Give an explanation though

File: asteriks_diamond_loop.c [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+int main() {
+int length; /*the user's input for the length of the half diamond*/
+printf("Write a positive integer for the length you want for half a diamond \n");
+scanf("%d", &length);
+int line;
+	for (line=1; line<=1; line++);
+		printf("* \n");
+	{
+		print("\n")
```
Comment (oldclesleycode) [inline asteriks_diamond_loop.c:14]: Compilation error - should be printf("\n");

File: asteriks_diamond_loop.c [hunk 7]
```diff
@@ -0,0 +1,23 @@
+/*Ava N.*/
+/*This has half a diamond on request of the user's preference of length*/
```
Comment (oldclesleycode) [inline asteriks_diamond_loop.c:2]: Hi Ava, your code neither compiled nor ran. When I fixed your compilation error and ran it, it still didn't properly run. Your code's logic is very off - you should see Ms. Hsu or me about this.

File: rps_game.c [hunk 8]
```diff
@@ -0,0 +1,103 @@
+/*Ava N.*/
+/*A game of rock, paper, scissors between the user and computer with a total of 5 rounds.*/
```
Comment (oldclesleycode) [inline rps_game.c:2]: Hi Ava, your code gave off 4 warnings. While it compiled, you should try and avoid these as much as possible. Your code also didn't go through five iterations of the game, so you missed out on a huge part of the assignment.

File: while_loop.c [hunk 9]
```diff
@@ -0,0 +1,16 @@
+/*Ava N.*/
```
Comment (oldclesleycode) [inline while_loop.c:1]: Nice job, but you forgot to include 2^0 = 1.

File: asteriks_diamond_loop.c (hunk 7 above)
Comment (AvaEN) [inline asteriks_diamond_loop.c:2]: Hi Lesley, I have been working through the problem and I understand it now. Have a nice weekend and see you on Tuesday! 
Ava

**Pull request closed without merging**